### PR TITLE
Komodo class warship

### PR DESCRIPTION
**Pull request by spockye**
## About The Pull Request

This adds the Komodo-Class Warship, a decent size ship used by the syndicate (Gorlex marauders in particular) for the war effort. 
image

Lore wise, the ship used to be a top of the line warship though as the war ended, and the Gorlex marauders became unable to sustain their warship, they were left with the orders to keep the ship operational at any cost.

 This leaves you with its current state. Due to the lack of funding, drastic measures were taken, such as scrapping the front artillery bays for cash, stripping the mech for parts and only keeping the ship barely functional.

when playing this ship, you are not meant to spec into research, or gaining better gear. you are to follow your standing orders, and restore the ships honour and integrity. the crew of the ship would still be following old protocol, and are to uphold Syndicate standards. 

TODO list:

- [x] - fix the bloody custom jobs
- [x] - finish the ruin version of the ship
- [x] - potential balance changes

 

## Why It's Good For The Game
more roleplay oriented ships, with challanges = good!

## Changelog
:cl:
add: adds a new ship(Komodo-class Syndicate warship
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/security_officer.dm [hunk 1]
```diff
@@ -278,3 +278,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 /obj/item/radio/headset/headset_sec/alt/department/sci
 	keyslot = new /obj/item/encryptionkey/headset_sec
 	keyslot2 = new /obj/item/encryptionkey/headset_sci
+
+	/datum/outfit/syndicate_empty/sbc/assault_komodo
```
Comment (Apogee-dev) [inline code/modules/jobs/job_types/security_officer.dm]: You accidentally nested this line underneath the previous block (starting with /obj/item/radio/headset/headset_sec/alt/department/sci). Delete a tab at the start of this line to fix the problem. You should also move this entire block up to before the "//Shiptest outfits end" comment in this file.

**Review by Apogee-dev — APPROVED** [review]

File: code/modules/jobs/job_types/security_officer.dm (hunk 1 above)
Comment (Apogee-dev) [reply code/modules/jobs/job_types/security_officer.dm]: Well, I meant to request changes, not approve, but still.

Comment (spockye) [reply code/modules/jobs/job_types/security_officer.dm]: That would potentially make the roles work?

**Comment by Valorium-RB** [comment]
Good looking ship!

File: code/modules/jobs/job_types/security_officer.dm (hunk 1 above)
Comment (Apogee-dev) [reply code/modules/jobs/job_types/security_officer.dm]: fixing the indentation absolutely will. Moving the code block up above the comment is just for the sake of organization.

Comment (spockye) [reply code/modules/jobs/job_types/security_officer.dm]: I'll have some time next sunday to get it fixed. If the bot marks this as stale somehow, im gonna hunt down its servers location and hurl insults at it

**Comment by Latentish** [comment]
I have no real particular complaints with how this is mapped. I really enjoy it and I think it's a prime example of ship design. However you happened to make a ship we basically already have. While I personally like this design more, the Gorlex Hyena is already a Syndicate assault ship. I would like this ship a lot more if you turned this into titanium hulls, nanotrasen weapons, and nanotrasen theming.

Edit: Regardless of theming, I am going to start asking all ships have a bathroom and running water somewhere.

**Comment by Apogee-dev** [comment]
> I have no real particular complaints with how this is mapped. I really enjoy it and I think it's a prime example of ship design. However you happened to make a ship we basically already have. While I personally like this design more, the Gorlex Hyena is already a Syndicate assault ship. I would like this ship a lot more if you turned this into titanium hulls, nanotrasen weapons, and nanotrasen theming.
> 
> Edit: Regardless of theming, I am going to start asking all ships have a bathroom and running water somewhere.

The Hyena is a salvage ship, not an assault ship, which is why it gets so much mining gear and has no actual security positions. It got a bit of a reputation as a combat ship primarily due to nerfs to Syndie equipment coming in much slower than expected (or not at all yet, in the case of hardsuits and softsuits). It's also something like half the size of the Komodo, so I don't really consider the two comparable.

**Comment by Latentish** [comment]
> The Hyena is a salvage ship, not an assault ship, which is why it gets so much mining gear and has no actual security positions. It got a bit of a reputation as a combat ship primarily due to nerfs to Syndie equipment coming in much slower than expected (or not at all yet, in the case of hardsuits and softsuits). It's also something like half the size of the Komodo, so I don't really consider the two comparable.

Yeah I suppose that's not a bad point. But I still feel like nanotrasen could use the dub more than syndicate. BUT it's extremely understandable that someone wouldn't want to redo it to this extent. It's a great ship still.

But I still require you to put a bathroom with a sink on the ship somewhere.

**Review by Latentish — APPROVED** [review]
This is really one of the better ships I've seen added recently. I want to see this testmerged. You really nailed it with this one.

File: code/modules/clothing/under/syndicate.dm [hunk 2]
```diff
@@ -272,3 +272,13 @@
 	. = ..()
 	var/obj/item/clothing/accessory/maidapron/inteq/A = new (src)
 	attach_accessory(A)
+
+/datum/outfit/syndicate/assult_komodo
```
Comment (Apogee-dev) [inline code/modules/clothing/under/syndicate.dm]: delet this outfit, it's in the wrong place and a copy of the one you actually want to be using

File: _maps/configs/komodo.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		"Medic":{
+			"outfit": "/datum/outfit/job/doctor/syndicate_komodo",
+			"slots": 1
+		},
+		"Mechanic": {
+			"outfit": "/datum/outfit/job/engineer/syndicate/gorlex",
+			"slots": 1
+		},
+		"Trooper":{
+			"outfit": "/datum/outfit/syndicate/assult_komodo",
```
Comment (Apogee-dev) [inline _maps/configs/komodo.json]: change this path to /datum/outfit/syndicate_empty/assult_komodo, and also correct that to "assault_komodo" to fix the typo.

**Comment by Derpguy3** [comment]
The power monitoring console in engineering doesn't seem to be hooked up to the grid, and isn't showing any draw or supply and rooms on the list.

**Comment by Latentish** [comment]
Please put a chair in front of the helm console.

File: code/modules/jobs/job_types/head_of_security.dm [hunk 4]
```diff
@@ -74,14 +74,14 @@
...
 	name = "Sergeant (Head of Security)"
 
-	uniform = /obj/item/clothing/under/syndicate/gorlex
-	head = /obj/item/clothing/head/HoS/beret/syndicate
+	uniform = /obj/item/clothing/under/syndicate/combat
+	head = /obj/item/clothing/head/warden
 	suit = /obj/item/clothing/suit/armor/vest/syndie
-	r_pocket = /obj/item/melee/transforming/energy/sword
-	backpack_contents = list(/obj/item/melee/baton/loaded=1)
+	id = /obj/item/card/id/syndicate_command/operative
```
Comment (Apogee-dev) [inline code/modules/jobs/job_types/head_of_security.dm]: replace this with /obj/item/card/id/syndicate_command/crew_id , the operative id doesn't actually display the character's real name. Additionally, make sure every job has this type of id if it doesn't inherit a syndie crew or captain id from its parent type.

**Review by Apogee-dev — CHANGES_REQUESTED** [review]

**Review by Latentish — APPROVED** [review]
Played this on live, didn't notice any issues. Going to marg this.

File: code/modules/jobs/job_types/security_officer.dm (hunk 1 above)
Comment (SweetBlueSylveon) [reply code/modules/jobs/job_types/security_officer.dm]: this was fixed

File: _maps/configs/komodo.json (hunk 3 above)
Comment (Latentish) [reply _maps/configs/komodo.json]: this was fixed

File: code/modules/jobs/job_types/security_officer.dm (hunk 1 above)
Comment (Latentish) [reply code/modules/jobs/job_types/security_officer.dm]: this was fixed

**Review by Apogee-dev — APPROVED** [review]
Forgot to approve this one. Looks great, has been working out pretty well in testing after you ironed out the initial issues.

**Pull request merged**